crypto/init.c

```diff
@@ -86,46 +90,6 @@ DEFINE_RUN_ONCE_STATIC(ossl_init_base)
     return 0;
 }
 
-static CRYPTO_ONCE register_atexit = CRYPTO_ONCE_STATIC_INIT;
-#if !defined(OPENSSL_SYS_UEFI) && defined(_WIN32)
-static int win32atexit(void)
-{
-    OPENSSL_cleanup();
-    return 0;
-}
-#endif
-
-DEFINE_RUN_ONCE_STATIC(ossl_init_register_atexit)
-{
-#ifndef OPENSSL_NO_ATEXIT
-# ifdef OPENSSL_INIT_DEBUG
-    fprintf(stderr, "OPENSSL_INIT: ossl_init_register_atexit()\n");
-# endif
-# ifndef OPENSSL_SYS_UEFI
-#  if defined(_WIN32) && !defined(__BORLANDC__)
-    /* We use _onexit() in preference because it gets called on DLL unload */
-    if (_onexit(win32atexit) == NULL)
```

Comment:
I'd still really like the understand what the situation is on Windows after this PR, for both shared and no-shared builds.

Reply:
I've performed the same tests which are implemented in [base64.so](https://github.com/Sashan/ossl-scripts/tree/main/base64.so) on windows too. Although  [windows version](https://github.com/Sashan/ossl-scripts/tree/main/base64.dll) requires a manual labor to walk through all cases covered in base64.so.